src/lib/field.ts

```diff
@@ -0,0 +1,1194 @@
+   */
+  static toAuxiliary(): [] {
+    return [];
+  }
+
+  /**
+   * This function is the implementation of {@link Provable.sizeInFields} for the {@link Field} type.
+   *
+   * Size of the {@link Field} type is 1, as it is the primitive type.
+   * This function returns a regular number, so you cannot use it to prove something on chain. You can use it during debugging or to understand the memory complexity of some type.
+   *
+   * @example
+   * ```ts
+   * console.log(Field.sizeInFields()); // Prints 1
+   * ```
+   *
+   * @return A number representing the size of the {@link Field} type in terms of {@link Field} type itself.
+   */
+  static sizeInFields() {
+    return 1;
+  }
+
+  /**
+   * Implementation of {@link Provable.fromFields} for the {@link Field} type.
+   *
+   * **Warning**: This function is designed for internal use. It is not intended to be used by a zkApp developer.
+   *
+   * Creates a {@link Field} from an array of Fields of length 1.
+   *
+   * @param fields - an array of length 1 serialized from {@link Field} elements.
+   *
+   * @return The first {@link Field} element of the given array.
+   */
+  static fromFields([x]: Field[]) {
+    return x;
+  }
+
+  /**
+   * This function is the implementation of {@link Provable.check} in {@link Field} type.
+   *
+   * As any field element can be a {@link Field}, this function does not create any assertions, so it does nothing.
+   *
+   * @param value - the {@link Field} element to check.
+   */
+  static check() {}
+
+  /**
+   * This function is the implementation of {@link Provable.toFields} for the {@link Field} type.
+   *
+   * The result will be always an array of length 1, where the first and only element equals the {@link Field} itself.
+   *
+   * @return A {@link Field} array of length 1 created from this {@link Field}.
+   */
+  toFields() {
+    return Field.toFields(this);
+  }
+
+  /**
+   * This function is the implementation of {@link Provable.toAuxiliary} for the {@link Field} type.
+   *
+   * As the primitive {@link Field} type has no auxiliary data associated with it, this function will always return an empty array.
+   */
+  toAuxiliary() {
+    return Field.toAuxiliary();
+  }
+
+  // ProvableExtended<Field>
+
+  /**
+   * Serialize the {@link Field} to a JSON string, e.g. for printing. Trying to print a {@link Field} without this function will directly stringify the Field object, resulting in an unreadable output.
+   *
+   * **Warning**: This operation does _not_ affect the circuit and can't be used to prove anything about the JSON string representation of the {@link Field}. Use the operation only during debugging.
+   *
+   * @example
+   * ```ts
+   * const someField = Field(42);
+   * console.log(someField.toJSON());
+   * ```
+   *
+   * @return A string equivalent to the JSON representation of the {@link Field}.
+   */
+  toJSON() {
```

Comment:
same as the note on the debug representation -- feel free to defer all this to another issue though, it's looking like it won't be a small change on top of what you have